docs/MUTATION_TESTING.md

```
@@ -0,0 +1,280 @@
+   - Replace `&&` with `||`, `>` with `>=`
+   - Replace function return value with default
+
+2. **Running the test suite**: Tests are executed against each mutated version
+
+3. **Measuring survival rate**:
+   - ✅ **Mutant killed**: Test failed → Good! Tests detected the artificial bug
+   - ❌ **Mutant survived**: Test passed → Bad! Tests did not catch the change
+
+### Mutation Score
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)